gordo/machine/dataset/datasets.py

```diff
@@ -240,32 +238,50 @@ def get_data(self) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
                 f"specified required threshold for number of rows ({self.n_samples_threshold})."
             )
 
+        if self.known_filter_periods:
+            data = pandas_filter_rows(data, self.known_filter_periods, buffer_size=0)
+            if len(data) <= self.n_samples_threshold:
+                raise InsufficientDataError(
+                    f"The length of the filtered DataFrame ({len(data)}) does not exceed the "
+                    f"specified required threshold for number of rows ({self.n_samples_threshold})"
+                    f" after dropping known periods."
+                )
+
         if self.row_filter:
             data = pandas_filter_rows(
                 data, self.row_filter, buffer_size=self.row_filter_buffer_size
             )
-
             if len(data) <= self.n_samples_threshold:
-                raise InsufficientDataAfterRowFilteringError(
-                    f"The length of the genrated DataFrame ({len(data)}) does not exceed the "
+                raise InsufficientDataError(
+                    f"The length of the filtered DataFrame ({len(data)}) does not exceed the "
                     f"specified required threshold for the number of rows ({self.n_samples_threshold}), "
-                    f" after applying the specified row-filter."
+                    f" after applying the specified numerical row-filter."
                 )
 
-        if self.low_threshold and self.high_threshold:
+        if isinstance(self.low_threshold, int) and isinstance(self.high_threshold, int):
+            assert (
+                self.low_threshold < self.high_threshold
+            ), "Low threshold need to be larger than high threshold"
```

Comment:
I think it is better to have this raised some custom exception as we did with `InsufficientDataError` just to keep in mind that in the future we will go to some  sort of custom retry policy logic, and in this case, we will just ignore some type of exceptions